Userland/Libraries/LibGfx/QOICommon.h

```diff
@@ -43,4 +45,35 @@ ALWAYS_INLINE static constexpr bool qoi_is_valid_run(u8 run)
     return AK::is_in_bounds(run, QOI_RUN_MIN, QOI_RUN_MAX);
 }
 
+class QOIState {
```

Comment:
`QOIState` seems overly generic for what this actually does. Please invent a better name?